core/domain/classifier_domain.py

```diff
@@ -387,37 +355,32 @@ def validate(self):
                     'Expected answers to be a list, received %s' %
                     grouped_answers['answers'])
 
-        # Classifier data can be either None (before its stored) or a dict.
-        if not isinstance(self.classifier_data, dict) and self.classifier_data:
-            raise utils.ValidationError(
-                'Expected classifier_data to be a dict|None, received %s' % (
-                    self.classifier_data))
-
-        if not isinstance(self.data_schema_version, int):
+        if not isinstance(self.algorithm_version, int):
             raise utils.ValidationError(
-                'Expected data_schema_version to be an int, received %s' %
-                self.data_schema_version)
+                'Expected algorithm_version to be an int, received %s' %
+                self.algorithm_version)
 
 
 class TrainingJobExplorationMapping(python_utils.OBJECT):
```

Comment:
StateTrainingJobsMapping

Reply:
Done. Also changed in `core/storage/classifier/gae_models`.